lib/logstash/codecs/telemetry.rb

```diff
@@ -258,26 +345,75 @@ def ts_codec_extract_path_key_value(path, data, filter_table, depth)
     end # branch, v iteration over hash passed down
   end # ts_codec_extract_path_key_value
 
+  private
+  def decode_kv_table(table,evs)
+
+    ev = Hash.new
+
+    if table[:timestamp]
+      ev[:timest] = Time.at(table[:timestamp]).to_datetime
+    end
+
+    datatypes = Array[:bytes_value,
+                 :string_value,
+                 :bool_value,
+                 :uint32_value,
+                 :uint64_value,
+                 :sint32_value,
+                 :sint64_value,
+                 :double_value,
+                 :float_value]
+
+    datatypes.each do |datatype|
+      if table.has_key?(datatype)
+        name = table[:name].to_s
+        if datatype == :bytes_value
+          value = table[datatype].to_s
+        else
+          value = table[datatype]
+        end
+        ev[name] = value
+      end
+    end
+
+    if table[:tables].length != 0
+      sub_tables = table[:tables]
+      evs_sub = Hash.new
+      sub_tables.each do |sub_table|
+        decode_kv_table(sub_table,evs_sub)
+      end
+      ev[:content] = evs_sub
+    else
+      @logger.debug? && @logger.debug("Iteration end")
```

Comment:
Not really enough context in this debug event. I would suggest that you either enhance it, or if enough content is captured when we unwrap and produce the gpb kv message, then just drop it.